src/util/find_symbols.h

```diff
@@ -79,12 +79,28 @@ void find_non_pointer_type_symbols(
   const exprt &src,
   find_symbols_sett &dest);
 
+/// Find identifiers with id ID_symbol of the sub expressions and the subs with
+/// ID in \p subs_to_find
+/// considering both free and bound variables, as well as any type tags.
+void find_type_and_expr_symbols(
+  const typet &src,
+  find_symbols_sett &dest,
+  const std::unordered_set<irep_idt> &subs_to_find);
```

Comment:
I don't think you need a `std::unordered_set` here. How about a `std::vector` or `std::list` (the former is likely even cheaper).